spec/directions/bake_figure_spec.rb

```diff
@@ -76,4 +106,14 @@
     end
   end
 
+  describe '#subfigure?' do
+    it 'can detect subfigures' do
+      is_subfigure = []
+      book2.chapters.figures.each do |figure|
+        is_subfigure.push(figure.subfigure?)
+      end
+      expect(is_subfigure).to eq([false, true, true])
```

Comment:
Is this an okay way to test this?

Reply:
Yes this is reasonable as part of the testing, but should collapse it to

```ruby
expect(book2.chapters.figures.map(&:subfigure?)).to eq([false, true, true])
```

`.map` does a `.each` accumulating results into an array for you.  Remember that `.map(&:subfigure?)` is Ruby shorthand for `.map{|figure| an_array_that_gets_returned.push(figure.subfigure?)}` which is shorthand for what you have above.

could even take out `.chapters` 

**My other question is, where do we test that recipe code like**

```ruby
chapter.figures.each do |figure|
  BakeFigure.v1(figure: figure,
                number: "#{chapter.count_in(:book)}.#{figure.count_in(:chapter)}")
end
```

is doing the right thing? like what is `count_in(:chapter)` for a subfigure?

Reply:
Just realized, should FigureElement method specs belong to a figure_element_spec file? We don't have one yet.